pkg/daemon/update.go

```diff
@@ -1594,6 +1599,26 @@ func (dn *Daemon) SetPasswordHash(newUsers, oldUsers []ign3types.PasswdUser) err
 	return nil
 }
 
+// Update the permission of the kubeconfig file located in /etc/kubenetes/kubeconfig
+// Requested in https://issues.redhat.com/browse/OCPBUGS-15367
+func (dn *Daemon) updateKubeConfigPermission() error {
+	klog.Info("updating the permission of the kubeconfig to: 0o600")
+
+	kubeConfigPath := "/etc/kubernetes/kubeconfig"
+	// Checking if kubeconfig is existed in the expected path:
+	if _, err := os.Stat(kubeConfigPath); err == nil {
+		if err := os.Chmod(kubeConfigPath, 0o600); err != nil {
+			return fmt.Errorf("Failed to reset permission for %s:%w", kubeConfigPath, err)
+		}
+	} else {
+		if !os.IsNotExist(err) {
+			return err
```

Comment:
nit: maybe format this error too, so we can find where this is happening later

Reply:
nit: The error returned by `os.Stat()` will indicate whether it was a file-not-found error or another error such as permissions. However, it appears that we're not doing anything special after detecting whether it was a file-not-found issue (which is fine in this case). Instead, it might be better to drop the `if !os.IsNotExist(err) {}` if block and return the formatted error from here, e.g.:

```golang
if _, err := os.Stat(kubeConfigPath); err == nil {
  // ...
} else {
  return fmt.Errorf("cannot stat %s: %w", kubeConfigPath, err)
}
```